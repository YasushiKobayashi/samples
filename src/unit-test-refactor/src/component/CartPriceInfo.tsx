import * as React from 'react'

import { ShippingFee } from './ShippingFee'

interface Props {
  shippingFee: number
  paymentAmount: number
}

export const CartPriceInfo: React.FC<Props> = ({ shippingFee, paymentAmount }) => {
  return (
    <div>
      <h2>金額情報</h2>
      <h3>送料</h3>
      <p>
        <ShippingFee shippingFee={shippingFee} />
      </p>
      <h3>合計金額</h3>
      <p>{`${paymentAmount.toLocaleString()}円`}</p>
    </div>
  )
}
