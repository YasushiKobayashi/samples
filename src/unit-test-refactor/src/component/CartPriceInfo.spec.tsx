import * as React from 'react'
import { cleanup, render } from '@testing-library/react'

import { CartPriceInfo } from './CartPriceInfo'

describe('CartPriceInfo', () => {
  afterEach(() => {
    cleanup()
  })

  it('1,1000円で送料無料', () => {
    const props = {
      shippingFee: 0,
      paymentAmount: 11000,
    }
    const { container, asFragment } = render(<CartPriceInfo {...props} />)

    expect(container.textContent).toContain('無料')
    expect(container.textContent).toContain('11,000円')
    expect(asFragment()).toMatchSnapshot()
  })
})
