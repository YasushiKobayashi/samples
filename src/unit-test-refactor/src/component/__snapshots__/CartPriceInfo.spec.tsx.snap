// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CartPriceInfo > 1,1000円で送料無料 1`] = `
<DocumentFragment>
  <div>
    <h2>
      金額情報
    </h2>
    <h3>
      送料
    </h3>
    <p>
      無料
    </p>
    <h3>
      合計金額
    </h3>
    <p>
      11,000円
    </p>
  </div>
</DocumentFragment>
`;

exports[`CartPriceInfo 1,1000円で送料無料 1`] = `
<DocumentFragment>
  <div>
    <h2>
      金額情報
    </h2>
    <h3>
      送料
    </h3>
    <p>
      無料
    </p>
    <h3>
      合計金額
    </h3>
    <p>
      11,000円
    </p>
  </div>
</DocumentFragment>
`;
